code/game/objects/items/weapons/medigun.dm

```diff
@@ -0,0 +1,389 @@
+		animate(color = "#00FF00", time = 2)
+		animate(color = "#0000FF", time = 2)
+
+		if(emagged)
+			var/chance_of_death = 80//"Most hearts couldn't whistand this voltage...
+
+			if(target_has_uberheart)
+				chance_of_death = 20
+
+			if(prob(chance_of_death))//...But I'm fairly certain your heart-"...
+				to_chat(ubertarget,"<span class='danger'>THIS--THIS IS TOO MUCH POWER")
+				spawn(160)
+					if(ubertarget)
+						ubertarget.flags &= ~INVULNERABLE
+						ubertarget.gib()
+			else
+				to_chat(ubertarget,"<span class='danger'>YOU ARE BULLETPROOF")
+				spawn(80)
+					if(ubertarget)
+						animate(ubertarget)
+						ubertarget.color = null
+						ubertarget.flags &= ~INVULNERABLE
+						if(heartcheck)
+							heartcheck.take_damage(rand(5,60))
+		else
+			var/duration = 40
+
+			if(target_has_uberheart)
+				duration = 60
+
+			spawn(duration)
+				if(ubertarget)
+					animate(ubertarget)
+					ubertarget.color = null
+					ubertarget.flags &= ~INVULNERABLE
+					if(heartcheck)
+						heartcheck.take_damage(rand(5,30))
+
+
+/obj/item/medigun/emag_act(mob/user as mob)
+	if(!emagged)
+		to_chat(user, "<span class='warning'>You swipe the cryptographic sequencer through the circuits.</span>")
+		emagged = 1
+		medigunpack.emagged = 1
+		medigunpack.update_icon()
+		icon_state = "medigunred"
+		item_state = "medigunred"
+		user.update_inv_back()
+		user.update_inv_hands()
+
+/obj/item/medigun/proc/heal(var/mob/living/M)
+	var/healing = MEDIHEAL
+	if(emagged)
+		healing = MEDIHARM
+
+	if(istype(M,/mob/living/carbon/human))
+		var/mob/living/carbon/human/H = M
+		var/datum/organ/internal/heart/heartcheck = H.internal_organs_by_name["heart"]
+		if(heartcheck.modifiers & OMODIFER_UBER)
+			healing = MEDIUBER
+
+	M.adjustOxyLoss(-2*healing)
+	M.heal_organ_damage(healing*2, 0)
+	M.heal_organ_damage(0, healing*2)
+	M.adjustToxLoss(-2*healing)
+	M.adjustCloneLoss(-2*healing)
+	M.adjustBrainLoss(-2*healing)
+	M.adjustBruteLoss(-2*healing)
+	M.adjustFireLoss(-2*healing)
+
+	if(healing > 0)
+		if(M.reagents)
+			var/list/reagents_to_check = list(
+				"toxin",
+				"stoxin",
+				"plasma",
+				"sacid",
+				"pacid",
+				"cyanide",
+				"amatoxin",
+				"chloralhydrate",
+				"carpotoxin",
+				"mindbreaker",
+				)
+			for(var/reagent in reagents_to_check)
+				if(M.reagents.has_reagent(reagent))
+					M.reagents.remove_reagent(reagent, healing)
+
+		if(istype(M,/mob/living/simple_animal))
+			var/mob/living/simple_animal/SA = M
+			if(SA.health > 0)
+				SA.health = min(SA.maxHealth,SA.health+2)
+
+		if(istype(M,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = M
+			for(var/datum/organ/external/temp in H.organs)
+				if(temp.status & ORGAN_BLEEDING)
+					temp.clamp()
+
+		if(M.losebreath >= 10)
+			M.losebreath = max(10, M.losebreath - 5)
+
+		if(M.bodytemperature > BODYTEMP_IDEAL)
+			M.bodytemperature = max(BODYTEMP_IDEAL, M.bodytemperature - (40 * TEMPERATURE_DAMAGE_COEFFICIENT))
+		else if(M.bodytemperature < (BODYTEMP_IDEAL+1))
```

Comment:
>Actually wait why, why aren't you just using an else here

because copypasted while i was tired. Changing that now. As well as where I copied it from.